hooks/pre-commit

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh -x
+#
+# Recommended pre-commit git hook for activity-stream github repo
+#
+# Install by executing
+#
+#   ln -s ../../hooks/pre-commit .git/hooks/pre-commit
+#
+# at the top-level of the activity-stream github repo.
+#
+# Runs `eslint --fix` on all selected files, which, given our current
+# prettier configuration, means prettifying these files as well.  The
+# commit will be aborted if eslint exits with a failure code.
+#
+# Based on the example script in the prettier docs at
+# https://prettier.io/docs/en/precommit.html
+
+FILES=$(git diff --cached --name-only --diff-filter=ACM "*.js" "*.jsx" "*.jsm" | sed 's| |\\ |g')
+echo "files to check: $FILES"
+[ -z "$FILES" ] && exit 0
+
+echo "$FILES" | xargs -t ./node_modules/.bin/eslint --cache --fix
+if [ $? -ne 0 ]
+then
+  echo "eslint failed; aborting commit.  Please fix and retry."
```

Comment:
You might want to say something like "eslint found errors but was unable to fix them all with --fix. Please check the output and attempt to manually fix" or along those lines

Reply:
Definitely better; will tweak.